test/linalg/triangular.jl

```diff
@@ -245,10 +250,12 @@ for elty1 in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
 
                 # Triangular-Triangular multiplication and division
                 elty1 != BigFloat && elty2 != BigFloat && @test_approx_eq full(A1*A2) full(A1)*full(A2)
-                @test_approx_eq full(A1'A2) full(A1)'full(A2)
-                @test_approx_eq full(A1*A2') full(A1)*full(A2)'
-                @test_approx_eq full(A1'A2') full(A1)'full(A2)'
-                @test_approx_eq full(A1/A2) full(A1)/full(A2)
+                if !(elty1 == Int && elty2 == Int)
```

Comment:
It is actually your change in 1016 that is causing the issue. When solving with `UnitXTriangular`s, you don't divide which the promotion rule these takes into account, but when you remove the `Unit` that is no longer true.